llvm/test/MC/AMDGPU/hsa-gfx12-v4.s

```diff
@@ -39,12 +39,12 @@
 // OBJDUMP-NEXT: 0080 00000000 00000000 00000000 00000000
 // OBJDUMP-NEXT: 0090 00000000 00000000 00000000 00000000
 // OBJDUMP-NEXT: 00a0 00000000 00000000 00000000 00000000
-// OBJDUMP-NEXT: 00b0 00000060 80000000 00000000 00000000
+// OBJDUMP-NEXT: 00b0 00000060 80000000 00040000 00000000
```

Comment:
This probably shouldn't have changed?

Reply:
This change is caused by the unintentional bug fix in this patch: the `AMDHSA_BITS_SET` macro is not encapsulated so expressions put into said macro may deviated from its intention: https://github.com/llvm/llvm-project/blob/e74bcecd36a59a9271c7d4133e73448a3def74c1/llvm/lib/Target/AMDGPU/Utils/AMDGPUBaseInfo.cpp#L1242-L1244

In the aforementioned case this means that the expansion will always result in 0. Now that the bits are set through MCExprs this expansion will be correct and means that, in this test case, the bit is now appropriately set.

Reply:
Can you fix this separately? Pre-committing converting the macro to use the do-while(0) pattern with a temp variable should work?